Assets/MixedRealityToolkit/Providers/BaseInputDeviceManager.cs

```diff
@@ -53,6 +53,100 @@ public abstract class BaseInputDeviceManager : BaseDataProvider<IMixedRealityInp
         /// <inheritdoc />
         public virtual IMixedRealityController[] GetActiveControllers() => System.Array.Empty<IMixedRealityController>();
 
+        private struct PointerConfig
+        {
+            public PointerOption profile;
+            public Stack<IMixedRealityPointer> cache;
+        }
+
+        private PointerConfig[] pointerConfigurations = System.Array.Empty<PointerConfig>();
+
+        private class PointerEqualityComparer : IEqualityComparer<IMixedRealityPointer>
```

Comment:
> private class PointerEqualityComparer : IEqualityComparer<IMixedRealityPointer> [](start = 7, length = 80)

One other thing to think about here (overall structure - I know that this is also another comment you're not interested in) - around how this could be written to be a bit more testable - I assume that some of these could be torn out into a different class which can be individually unit/component tested, which MIGHT help.

I say might because I haven't actually done the deep thinking to know if it's actually valuable/not valuable, as it's possible that this level of testing isn't valuable, but it could be!

Reply:
> I assume that some of these could be torn out into a different class which can be individually unit/component tested, which MIGHT help.

@wiwei  I'm not exactly understanding you. What are "these" that could be torn out?